src/domain/usecase/CreateSpot.test.ts

```diff
@@ -0,0 +1,103 @@
+import { CreateSpot } from './CreateSpot'
+import { SpotRepository } from '../repository/SpotRepository'
+import { Spot, CreateSpotCommand } from '../model/Spot'
+import { Location } from '../model/Location'
+
+const createSpot = (createSpotCommand: CreateSpotCommand): Spot => ({
+  id: 'spot-id',
+  location: {
+    address: 'address',
+    latitude: 0.1,
+    longitude: 0.2,
+  },
+  name: 'spot name',
+  description: 'spot description',
+})
+
+describe('GetSpot', () => {
+  let repository: SpotRepository
+  let useCase: CreateSpot
+
+  beforeEach(() => {
+    repository = {
+      getSpot: jest.fn(),
+      getSpotsByLocation: jest.fn(),
+      createSpot: jest.fn(),
+    }
+    useCase = new CreateSpot({ spotRepository: repository })
+  })
+
+  it('execute: should create the spot with name and location', async () => {
+    // Given
+    const name = 'spot name'
+    const location: Location = {
+      latitude: 1,
+      longitude: 1,
+    }
+    ;(repository.createSpot as jest.Mock).mockResolvedValue(
+      createSpot(new CreateSpotCommand(name, location))
+    )
+
+    // When
+    const result: any = await useCase.execute(name, location)
+
+    // Then
+    expect(repository.createSpot).toHaveBeenCalledTimes(1)
+    expect(repository.createSpot).toHaveBeenCalledWith({ name, location })
+    expect(result.data).toEqual(createSpot({ name, location }))
```

Comment:
I'm not too sure about this pattern. Is is testing something?

Reply:
Yep as long as `createSpot` is implemented correctly :)